shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -366,6 +366,30 @@ public View getPlatformViewById(Integer id) {
         return controller.getView();
     }
 
+    /**
+    * Callback fired when the platform's input connection is locked, or should be used. See also
+    * {@link TextInputPlugin#lockPlatformViewInputConnection}.
+    */
+    public void onInputConnectionLocked(int viewId) {
```

Comment:
instead of adding these methods and triggering this from TextInputPlugin we could directly trigger this from resizePlatformView.

Reply:
I like having to avoid the extra plumbing, but I don't like having the actual call back into the plugin be separated from the real bit in `TextInputPlugin`. Moving the logic there feels brittle to me, like one could get called without the other and then things would end up out of sync. WDYT? I could be being overly cautious.

Reply:
I see your point, I do not expect there to ever be another call site for lock/unlock so my personal preference would be to avoid the plumbing at the cost of not having a mechanism enforcing the coupling of these calls.

Reply:
IMO, bypassing TextInputPlugin for a text input related behavior sounds like an architectural issue waiting to happen. I think TextInputPlugin should be involved in all behaviors that impact Android text input related to a Flutter experience. I don't think we can say with confidence where any such requirements will go in the future....